### ISSUE TITLE
applyPatch returns error when source prop is undefined

### ISSUE THREAD

**Issue by ulisses-alves** [chbrown/rfc6902#40]
Hi there,

Although `createPatch` apparently creates the correct patch for the following operation, `applyPatch` fails with `Value required at path: /value`

```javascript
var rfc6902 = require("rfc6902")

var orig = { value: undefined }
var modified = { value: 1 }
var patch = rfc6902.createPatch(orig, modified)
rfc6902.applyPatch(orig, patch)
```

Is that expected behavior?

Thanks in advance.

**Comment by chbrown** [comment]
Duplicate of #35. Good reminder that I need to deploy a new version that includes that fix, though!

**Comment by ulisses-alves** [comment]
Is there any side effect of switching all operations' `op` from `replace` to `add` before applying a patch? I just to make it work until you get fix published.

**Comment by chbrown** [comment]
Yeah, that should work — as long as the target isn't an Array, the `add` operation works just like `replace`, but without the existence check (which is what's breaking in v2.4.0).

**Comment by chbrown** [comment]
Just released `3.0.0` – sorry for the delay and consequent hacks you had to make!
